pkgs/development/cuda-modules/_cuda/lib/db.nix

```diff
@@ -0,0 +1,200 @@
+{ _cuda, lib }:
+let
+  cudaLib = _cuda.lib;
+  inherit (cudaLib) selectQuantized matchQuantized;
+in
+{
+  _archiveIsAvailableOn =
+    db: hostPlatform: sha256:
+    let
+      archive = db.archive.sha256.${sha256};
+      inherit (archive) systemNv;
+      # NOTE: We haven't moved `cudaCapabilities` &co to the platform yet, so we expect the user to `//`-merge the attribute to `hostPlatform`
+      isJetson = hostPlatform.isJetson or false;
+      systemStringMatches =
+        (isJetson -> db.system.isJetson.${systemNv} or false)
+        && lib.hasAttr db.system.fromNvidia.${systemNv} hostPlatform.system;
+      constraintsSatisfied = builtins.all lib.id (
+        pname: constraints: lib.mapAttrsToList (sign: version: false)
+      ) archive.extraConstraints;
```

Comment:
This doesn't seem correct -- `builtins.all` is oversaturated, `lib.mapAttrsToList` is undersaturated, etc.